woocommerce/woocommerce-android
# Update orders on notification

Closes: #11794

### Description
This PR replaces the outdated EventBus-based order data refresh mechanism with a more reliable and efficient WorkManager task.

Key changes:

- WorkManager Integration: Introduced a WorkManager task to handle order data updates.
- Order Fetching: Fetches order information from the notification.
- Timestamp Update: Updates the last update timestamp.

By utilizing WorkManager, we enhance the robustness and predictability of order data updates, improving overall application performance and reliability.

### Testing information
1. Open the app
2. Go to the orders section
3. On the web, open the store and create a new order
4. Check that when the notification is received, the app updates the list of orders with the newly received order
5. Check that the last update timestamp is updated

### Images/gif

https://github.com/user-attachments/assets/ecf2fb49-0166-441c-a61e-442da2ea0d0b
